crates/red_knot/src/types.rs

```diff
@@ -336,6 +380,35 @@ impl FunctionTypeId {
     }
 }
 
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct ModuleTypeId {
+    module: Module,
+    file_id: FileId,
```

Comment:
Uh interesting. I assumed that `Module` would store the `FileId` but it indeed stores the `PathBuf`. I wonder if we should change `Module` to store a `FileId` instead of the `Path`. But I need to think this through more carefully before we do it.

Reply:
Is there any change required for `ModuleTypeId` here? :)

Reply:
I think it's fine as is for now. We can always change later.